nncf/tensorflow/sparsity/magnitude/algorithm.py

```diff
@@ -190,7 +190,7 @@ def _select_threshold(self, sparsity_level):
         if not all_weights:
             return 0.0
         all_weights_tensor = tf.sort(tf.concat(all_weights, 0))
-        index = int(tf.cast(tf.size(all_weights_tensor), all_weights_tensor.dtype) * sparsity_level)
+        index = int(tf.cast(tf.size(all_weights_tensor) - 1, all_weights_tensor.dtype) * sparsity_level)
```

Comment:
Fixed a minor bug. Not consistent with PT [implementation ](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/sparsity/magnitude/algo.py#L145)